mgr/manager.go

```diff
@@ -31,6 +31,59 @@ type CertServiceManager struct {
 	serviceManager svcmgr.Manager
 }
 
+// Process a spec updating content on disk, taking action as needed.
+// Returns (TTL for PKI, error).  If an error occurs, the ttl is at best
+// a hint to the invoker as to when the next refresh is required- that said
+// the invoker should back off and try a refresh.
+func (csm *CertServiceManager) EnforcePKI(enable_actions bool) (time.Duration, error) {
+	err := csm.CheckDiskPKI()
+	if err != nil {
+		log.Debugf("manager: spec %s, checkdiskpki: %s.  Forcing refresh.", csm, err.Error())
+		csm.ResetLifespan()
+	}
+
+	if err = csm.CheckCA(); err != nil {
+		log.Errorf("manager: the CA for %s has changed, but the service couldn't be notified of the change", csm.Spec.Path)
+	}
+
+	lifespan := time.Duration(0)
+	if !csm.Ready() {
+		log.Infof("manager: queueing %s because it isn't ready", csm.Spec.Path)
+	} else {
+		log.Debugf("manager: %s checking lifespan", csm.Spec.Path)
+		lifespan = csm.Lifespan()
+		if lifespan <= 0 {
+			log.Info("manager: queueing certificate with lifespan of ", lifespan.Hours(), " hours")
```

Comment:
Same here -- why do we have a log statement here since we renew PKI if the lifespan <= 0 below?

Reply:
Removed this; this was primarily for my own log verbosity/debugging, but it's fine to remove.